recipe/thirdparty/providers/boxysaml.go

```diff
@@ -51,7 +52,7 @@ func BoxySaml(input tpmodels.ProviderInput) *tpmodels.TypeProvider {
 				config.UserInfoEndpoint = boxyURL + "/api/oauth/userinfo"
 			}
 
-			return config, err
+			return discoverOIDCEndpoints(config)
```

Comment:
Should not be called in here.. the ksjdfnio38oi function should only be called in one place. Which is in the custom.go file's `GetConfigForClientType`. We need to change how the code is structured.. yet again.